src/neo4j_genai/components/pdf_loader.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import io
+from abc import abstractmethod
+from pathlib import Path
+from typing import Optional
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.implementations.local import LocalFileSystem
+
+from neo4j_genai.exceptions import PdfLoaderError
+from neo4j_genai.pipeline import Component, DataModel
+
+
+class PdfDocument(DataModel):
```

Comment:
Maybe a more generic name since text might come from something else than PDF? `Document`?

Reply:
I was thinking we can have different document types inherit from a base `Document` when we add other types of data, I didn't want to optimize for that in this PR. What do you think?

Reply:
Fair enough :)